component/attr_cache/cacheMap.go

```diff
@@ -81,13 +144,20 @@ func (value *attrCacheItem) isInCloud() bool {
 func (value *attrCacheItem) markDeleted(deletedTime time.Time) {
 	value.attrFlag.Clear(AttrFlagExists)
 	value.attrFlag.Set(AttrFlagValid)
+	value.attrFlag.Set(AttrFlagNotInCloud)
 	value.cachedAt = deletedTime
 	value.attr = &internal.ObjAttr{}
+	for _, val := range value.children {
```

Comment:
To improve performance, this for loop should only run if the item being marked deleted was not already marked deleted. This matters especially when renaming a directory with lots of depth to it, because we mark successive levels of directories deleted, which would cause a lot of repeated tree traversal here.